scripts/forward.py

```diff
@@ -2,90 +2,122 @@
+# both height and width must be divisible by 8
+img_height = 192
+img_width = 256
 
-def rescale(img, height, width):
-    if len(img.shape) == 3:
-        h, w, _ = img.shape
-    elif len(img.shape) == 2:
-        h, w = img.shape
+def rescale_image(img, tar_h, tar_w):
+    img_h, img_w = img.shape[:2]
 
-    if (float(height)/h) > (float(width)/w):
-        img = skimage.transform.resize(img, (height, w*height/h),
-                                       order=3, preserve_range=True)
+    if img_h > tar_h or img_w > tar_w:
+        interpolation = cv.INTER_AREA
     else:
-        img = skimage.transform.resize(img, (h*width/w, width),
-                                       order=3, preserve_range=True)
+        interpolation = cv.INTER_CUBIC
 
-    if len(img.shape) == 3:
-        h, w, _ = img.shape
-    elif len(img.shape) == 2:
-        h, w = img.shape
+    aspect_ratio = img_w / img_h
+    target_ratio = tar_w / tar_h
+    height_ratio = tar_h / img_h
+    width_ratio = tar_w / img_w
+
+    new_h, new_w = tar_h, tar_w
 
-    img = img[h//2-(height/2):h//2+(height/2), w//2-(width/2):w//2+(width/2)]
+    if aspect_ratio != target_ratio:
+        if width_ratio > height_ratio:
+            new_w = np.floor(new_h * aspect_ratio)
+        elif width_ratio < height_ratio:
+            new_h = np.floor(new_w / aspect_ratio)
+
+    rescaled_img = cv.resize(img, (int(new_w), int(new_h)),
+                             interpolation=interpolation)
+    return rescaled_img
+
+
+def pad_image(img, tar_h, tar_w):
+    img_h, img_w = img.shape[:2]
 
     if len(img.shape) == 3:
-        img[:, :, 0] -= 103.939
-        img[:, :, 1] -= 116.779
-        img[:, :, 2] -= 123.68
+        pad_value = [126]*3
+    else:
+        pad_value = 0
+
+    pad_vert = np.abs(tar_h - img_h) / 2
+    pad_horz = np.abs(tar_w - img_w) / 2
+
+    pad_t = np.floor(pad_vert).astype(int)
+    pad_b = np.ceil(pad_vert).astype(int)
+    pad_l = np.floor(pad_horz).astype(int)
+    pad_r = np.ceil(pad_horz).astype(int)
+
+    padded_img = cv.copyMakeBorder(img, pad_t, pad_b, pad_l, pad_r,
+                                   borderType=cv.BORDER_CONSTANT,
+                                   value=pad_value)
+    return padded_img
 
-    return img
 
 class Saliency():
     def __init__(self):
-        model_file = rospy.get_param('~saliency_model_file', '/tmp/model.hdf5')
-        weight_file = rospy.get_param('~saliency_weight_file', '/tmp/weights.hdf5')
-
-        if (not os.path.exists(model_file) or not os.path.exists(weight_file)):
-            rospy.logwarn("Model files not present:\n\t{}\n\t{}\nWe will download them from owncloud."
-                          .format(model_file, weight_file))
-            import wget
-            import ssl
-            ssl._create_default_https_context = ssl._create_unverified_context
-            wget.download("https://neurorobotics-files.net/owncloud/index.php/s/rnU4XXGGVowynfQ/download", model_file)
```

Comment:
The new URLs are there:
https://neurorobotics-files.net/owncloud/index.php/apps/files?dir=/general-files/CDP4%20saliency%20model/tensor-flow&fileid=85077